File: TekstEditorP/Tekst.py
from tkinter import Toplevel, Menu, Text, Label
from tkinter.filedialog import asksaveasfilename
from ZoekDialoog import ZoekDialoog

class Tekst(Toplevel):
    def __init__(self,parent):
        super().__init__(parent)
        self.geometry("500x300")

        menubar = Menu(self)
        filemenu = Menu(menubar, tearoff=0)
        filemenu.add_command(label="Opslaan",        command=self.opslaan)
        filemenu.add_command(label="Opslaan als...", command=self.opslaanAls)
        filemenu.add_command(label="Sluiten",        command=self.destroy)
        menubar.add_cascade(label="File", menu=filemenu)
        filemenu = Menu(menubar, tearoff=0)
        filemenu.add_command(label="Zoek",    command=self.zoek)
        filemenu.add_command(label="Vervang", command=self.vervang)
        menubar.add_cascade(label="Zoek", menu=filemenu)
        self.configure(menu=menubar)

        self.invoer = Text(self, width=1, height=1)
        self.invoer.pack(fill="both", expand=True)
        self.invoer.bind("<KeyRelease>", self.verander)

        self.status = Label(self)
        self.status.pack()

    def opslaan(self):
        if self.title():
            self.schrijfNaarFile()
        else:
            self.opslaanAls()
        
    def opslaanAls(self):
        naam = asksaveasfilename()
        if naam:
            self.title(naam)
            self.schrijfNaarFile()

    def schrijfNaarFile(self):
        naam = self.title()
        schrijver = open(naam, "w")
        schrijver.write(self.invoer.get("1.0", "end"))
        schrijver.close()
        pass
    
    def leesVanFile(self, naam):
        lezer = open(naam, "r")
        self.invoer.insert("1.0", lezer.read())
        lezer.close()
        self.title(naam)
        self.verander(None)

    def verander(self, ea):
        self.status.configure(text=str(len(self.invoer.get("1.0", "end")))+" karakters")

    def zoekOfVervang(self, ookVervangen):
        dialoog = ZoekDialoog(self, ookVervangen)
        self.wait_window(dialoog)
        if dialoog.ZoekOK:
            zoek = dialoog.ZoekText
            beginpos = self.invoer.search(zoek, "1.0")
            if beginpos:
                eindpos = f"{beginpos}+{len(zoek)}c"
                if ookVervangen:
                    vervang = dialoog.VervangText
                    self.invoer.delete(beginpos, eindpos)
                    self.invoer.insert(beginpos, vervang)
                    eindpos = f"{beginpos}+{len(vervang)}c"
                    self.verander(None)

                self.invoer.tag_add("sel", beginpos, eindpos)
                self.invoer.mark_set("insert", eindpos)

    def zoek(self):
        self.zoekOfVervang(False)

    def vervang(self):
        self.zoekOfVervang(True)

File: TekstEditorC/Tekst.cs
using System;
using System.Drawing;
using System.IO;
using System.Windows.Forms;
public class Tekst : Form
{
    TextBox invoer;
    Label status;

    public Tekst()
    {
        MenuStrip menuStrip = new MenuStrip();
        invoer = new TextBox();
        status = new Label();

        ToolStripDropDownItem menu = new ToolStripMenuItem("File");
        menu.MergeAction = MergeAction.MatchOnly;
        menuStrip.Items.Add(menu);
        menu.DropDownItems.Add("Opslaan", null, opslaan);
        menu.DropDownItems.Add("Opslaan &als...", null, opslaanAls);
        menu.DropDownItems.Add("Sluiten", null, sluiten);
        menu = new ToolStripMenuItem("Zoek");
        menu.DropDownItems.Add("Zoek", null, zoek);
        menu.DropDownItems.Add("Vervang", null, vervang);
        menuStrip.Items.Add(menu);
        menuStrip.Visible = false;
        Controls.Add(menuStrip);

        invoer.Multiline = true;
        invoer.WordWrap = false;
        invoer.ScrollBars = ScrollBars.Both;
        invoer.Font = new Font("Courier New", 12);
        invoer.TextChanged += verander;
        Controls.Add(invoer);
        Controls.Add(status);

        ClientSize = new Size(500, 300);
        Resize += vergroot;
        vergroot(null, null);
    }
    private void vergroot(object o, EventArgs ea)
    {
        invoer.Location = new Point(10, 10);
        invoer.Size = ClientSize - new Size(20, 30);
        status.Location = new Point(10, ClientSize.Height - 15);
    }
    private void verander(object o, EventArgs ea)
    {
        status.Text = invoer.Text.Length.ToString() + " karakters";
    }
    private void sluiten(object sender, EventArgs e)
    {
        Close();
    }
    private void zoekOfVervang(bool ookVervangen)
    {
        ZoekDialoog dialoog = new ZoekDialoog(ookVervangen);
        if (dialoog.ShowDialog(this) == DialogResult.OK)
        {
            string alles = invoer.Text;
            string zoek = dialoog.ZoekText.Text;
            int pos = alles.IndexOf(zoek);
            if (pos >= 0)
            {
                if (ookVervangen)
                {
                    string vervang = dialoog.VervangText.Text;
                    invoer.Text = alles.Replace(zoek, vervang);
                    zoek = vervang;
                }
                invoer.Select(pos, zoek.Length);
            }
        }
    }
    private void zoek(object o, EventArgs ea)
    {
        zoekOfVervang(false);
    }
    private void vervang(object o, EventArgs ea)
    {
        zoekOfVervang(true);
    }
    private void opslaan(object o, EventArgs ea)
    {
        if (Text == "")
            opslaanAls(o, ea);
        else schrijfNaarFile();
    }
    private void opslaanAls(object o, EventArgs ea)
    {
        SaveFileDialog dialoog = new SaveFileDialog();
        dialoog.Filter = "Tekstfiles|*.txt|Alle files|*.*";
        dialoog.Title = "Tekst opslaan als...";
        if (dialoog.ShowDialog() == DialogResult.OK)
        {
            Text = dialoog.FileName;
            schrijfNaarFile();
        }
    }
    private void schrijfNaarFile()
    {
        StreamWriter writer = new StreamWriter(Text);
        writer.Write(invoer.Text);
        writer.Close();
    }
    public void LeesVanFile(string naam)
    {
        StreamReader reader = new StreamReader(naam);
        invoer.Text = reader.ReadToEnd();
        reader.Close();
        Text = naam;
    }
}